core/math/quick_hull.cpp

```diff
@@ -292,7 +292,7 @@ Error QuickHull::build(const Vector<Vector3> &p_points, Geometry3D::MeshData &r_
 
 		//distribute points into new faces
 
-		for (List<Face>::Element *&F : lit_faces) {
+		for (List<Face>::Element *F : lit_faces) {
```

Comment:
Changed this, this code with nested lists is a bit weird but I don't think `*&` brings anything compared to `*`?

Reply:
I think the new code is creating a copy of the pointer on each iteration. It's probably not to bad on performance, especially on release builds, but if we want to avoid that copy we need the `&`.

I haven tested, but  if we want to be consistent with the other `for`sit should be something like `const List<Face>Element *&F`. That's a bit of a mouthful, so I don't know if we want to go with that or just accept that we may be copying the pounters.

Reply:
Also, we may be able to add parentheses to make the type clear. We need to test, but I feel something like `const (List<Face>::Element*) &F` should work just the same.

Reply:
That's a good point indeed.

For the reference, the original code before #50511 was:
```
for (List<List<Face>::Element *>::Element *F = lit_faces.front(); F; F = F->next()) {
```

I'll leave the `*&foo` occurrences be for this PR, we can see in a second pass if we actually use them for a good reason or if they should be simplified.